src/generated/resources/data/forge/tags/worldgen/biome/is_slope.json

```diff
@@ -0,0 +1,10 @@
+{
+  "replace": false,
+  "values": [
+    "minecraft:savanna_plateau",
+    "minecraft:wooded_badlands",
```

Comment:
do plateaus and wooded badlands belong in slopes? I see slopes as the sides of the giant mountainous areas that end with peak biomes in center. Savannas plateau and badlands are more their own things separate from that mountainous area. Here's a pic of what I mean. You can see how these mountains have a pretty clear area of what is a slope for them around the peaks and lower down. Basically anywhere that biomes can spawn alongside Snowy Slopes biome
![image](https://user-images.githubusercontent.com/40846040/163280575-834638f5-f40c-4bba-a5fb-3f63b3ee9014.png)

Reply:
Savanna plateaus and wooded badlands were added to slopes with the forge biomedict tags because they do have the possibility to generate as slopes. If you can find an area like that screenshot in warmer biomes you should run into plateaus in place of the cold slope biomes. The OverworldBiomeBuilder class shows this.

Reply:
Leaving my final 2 cents over the slopes tag here. The slopes tags is for biomes that are dedicated to spawning around the main body of giant peaking mountains like meadows, snowy slopes, and groves do. these biomes spawn nowhere else except for those areas. Savanna Plateaus spawn frequently throughout the savanna area and often nowhere near any giant peaking mountain. Thus they are plateaus and not slopes. Wooded Badlands do seem to be spawning as mountains themselves or higher plateaus. I'll concede that wooded badlands can stay in the slopes tag but they are a trickier biome since they do the peak, slope, and plateau areas all together.

![image](https://user-images.githubusercontent.com/40846040/163287754-6c3c7277-7410-45e7-8d36-b520b5bed5ec.png)
![Untitled](https://user-images.githubusercontent.com/40846040/163287758-3ce6fc6a-57c5-4312-be47-fb43c807b584.png)

Still, one has to think of the usability of the tag as well. Worldgen structures json can only be given one structure to spawn in and other usages of these tags may also be highly tailored to the slopes of the main giant mountains everyone thinks about with regards to groves and snowy slopes and meadows. Having savanna and/or badlands biomes would cause mods using the slopes to have their stuff be out of place and not designed to fit in a badland's area. It's up to the PR maintainer to decide what to do but if anything, adding a cold_slopes, cold_peaks, warm_slopes, warm_peaks may be beneficial in the long run to help people target the right temperature mountain. Although, savanna plateau should probably stay only in the plateau and savanna tag.

Reply:
Firstly, I should state that meadows and savanna have identical elevation spawning, so if you want savanna plateaus removed from the slope tag you should also remove the meadows, leaving the slope tag dedicated for the two slope only biomes, the snowy slopes and the grove.

However, I think both should stay in the slope tag as well as the plateau tag as they can spawn in both areas, to put it simply. If they shouldn't be in the slope tag because they don't spawn on slopes 100% of the time then you should remove them from the plateau tag as well surely, as they don't spawn as plateaus 100% of the time either. Even pure slope biomes, the grove and the snowy slope, don't always generate next to mountains (if the erosion noise didn't peak high enough). I definitely think it's better to put them in tags for areas that they _can_ spawn rather than put them in tags for areas that are the only place they can spawn.

Some tag changes of some sort are probably needed in the future tho, since the functionality possible with biome dictionary tags doesn't seem to be easily translatable into the vanilla system.